docs/content/v0.20/architecture.md

```diff
@@ -62,21 +63,21 @@ to find objects that are synced by the OCM agent, gets their status
 and updates *WorkStatus* objects in the ITS namespace associated with the WEC.
 
 <figure>
-  <img src="./main-modules.png"  alt="Main Modules">
+  <img src="./images/main-modules.png"  alt="Main Modules">
   <figcaption align="center">Figure 2 - Main Modules </figcaption>
 </figure>
 
 ## KubeStellar Controller Manager
 
-This module manages the placement and status controllers. The placement controller watches 
-placement and workload objects on the Workload Definition Space 
-(WDS) and wraps workload objects into a manifest for delivery through the 
-Inventory and Transport Space (ITS). The status controller watches for *WorkStatus* objects on the 
-ITS and updates the status of objects in the WDS when singleton status is requested 
-in the placement for those objects. There is one instance of a KubeStellar Controller 
-Manager for each WDS. Currently this controller-manager runs in the KubeFlex hosting cluster and 
-is responsible of installing the required CRDs in the associated WDS.
-More details on the internals of this module are provided in [KubeStellar Controllers Architecture](#kubestellar-controllers-architecture)
+This module manages binding controller (a.k.a placement controller before the rename mentioned at the very beginning of this doc) and status controller. The binding controller watches `BindingPolicy` and workload objects on the Workload Definition Space (WDS) and wraps workload objects into a manifest for delivery through the Inventory and Transport Space (ITS). The status controller watches for *WorkStatus* objects on the ITS and updates the 
+status of objects in the WDS when singleton status is requested in the `BindingPolicy` for those objects. 
+There is one instance of a KubeStellar Controller Manager for each WDS. Currently this controller-manager runs in the KubeFlex hosting cluster and is responsible of installing the required CRDs in the associated WDS.
+More details on the internals of this module are provided in [KubeStellar Controllers Architecture](#kubestellar-controllers-architecture).
+
+### Planned changes
+
+As a step towards a broader change we're currently working on, we plan to decouple the transport handling from the binding controller and to separate them into two different controllers (binding controller & transport controller).
+As part of this change, current binding controller also creates from the `BindingPolicy` a matching `Binding` object in the WDS which contains the concrete workload objects and concrete list of clusters that were selected by the `BindingPolicy`. The future transport controller should watch `Binding` and get from the WDS api server the objects that are specified in the workload section, wrap them into a wrapped object and deliver the wrapped object through the ITS to the WECs specifid in the `Binding` destinations. 
```

Comment:
The `Binding` does not "contain" the workload objects, it contains _references to_ the workload objects.

Reply:
"WDS api server(s)"